### PR TITLE
Display multiple DOIs

### PR DESCRIPTION
**Pull request by monaawi**
* INSPIR-2238
* Refactored ListItemAction to be more generalised, it now simply applies styling to the component it receives

### CODE REVIEW

File: ui/src/common/components/ListItemAction/__tests__/ListItemAction.test.jsx [hunk 1]
```diff
@@ -2,50 +2,18 @@ import React from 'react';
 import { shallow } from 'enzyme';
 import { Button } from 'antd';
 
+import IconText from '../../IconText';
 import ListItemAction from '../ListItemAction';
 
 describe('ListItemAction', () => {
-  it('renders with link (href and target)', () => {
+  it('renders with child button', () => {
```
Comment (harunurhan) [inline ui/src/common/components/ListItemAction/__tests__/ListItemAction.test.jsx]: simply`renders` is enough

File: ui/src/common/components/ListItemAction/__tests__/ListItemAction.test.jsx [hunk 2]
```diff
@@ -2,50 +2,18 @@ import React from 'react';
...
-      <ListItemAction
-        iconType="info"
-        text="Test"
-        link={{
-          href: '//example.com',
-          target: '_blank',
-        }}
-      />
+      <ListItemAction>
+        <Button onClick={undefined}>
```
Comment (harunurhan) [inline ui/src/common/components/ListItemAction/__tests__/ListItemAction.test.jsx]: `onClick={undefined}` ?

File: ui/src/literature/components/DOILinkAction.jsx [hunk 3]
```diff
@@ -10,14 +12,11 @@ class DOILinkAction extends Component {
     const href = `//doi.org/${doi}`;
     return (
       <EventTracker eventId="DoiLink">
```
Comment (harunurhan) [inline ui/src/literature/components/DOILinkAction.jsx]: now all even trackers should directly wrap `Button` or `Link` whatever is inside.

File: ui/src/literature/components/LiteratureItem.jsx [hunk 4]
```diff
@@ -57,11 +58,18 @@ class LiteratureItem extends Component {
...
-                <ListItemAction
-                  iconType="login"
-                  text={`${citationCount} citations`}
-                  link={{ to: `${LITERATURE}?q=refersto:recid:${recordId}` }}
-                />
+                <ListItemAction>
+                  <Link
+                    className="no-transition ant-btn"
+                    to={`${LITERATURE}?q=refersto:recid:${recordId}`}
+                    onClick={undefined}
```
Comment (harunurhan) [inline ui/src/literature/components/LiteratureItem.jsx]: ?

File: ui/src/literature/components/__tests__/CiteModalAction.test.jsx [hunk 5]
```diff
@@ -26,7 +26,10 @@ describe('CiteModalAction', () => {
     const setCiteContentFor = jest.fn();
     wrapper.instance().setCiteContentFor = setCiteContentFor;
     wrapper.update();
-    const onCiteClick = wrapper.find(ListItemAction).prop('onClick');
+    const onCiteClick = wrapper
+      .find(ListItemAction)
+      .find(Button)
```
Comment (harunurhan) [inline ui/src/literature/components/__tests__/CiteModalAction.test.jsx]: can you try `prop('children')`

File: ui/src/common/components/DropdownMenu.jsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        <span>{title}</span>
+      </Dropdown>
+    );
+  }
+}
+
+DropdownMenu.propTypes = {
+  children: PropTypes.node.isRequired,
+  title: PropTypes.node.isRequired,
+  onClick: PropTypes.func,
```
Comment (harunurhan) [inline ui/src/common/components/DropdownMenu.jsx]: `// eslint-disable-line <rule_name>` instead of putting defaultProps

File: ui/src/literature/components/ArxivPdfDownloadAction.jsx [hunk 7]
```diff
@@ -1,24 +1,23 @@
...
-          text="pdf"
-          link={{
-            href,
-            target: '_blank',
-          }}
-        />
-      </EventTracker>
+      <ListItemAction>
+        <EventTracker eventId="PdfDownload">
+          <Button href={href} target="_blank">
```
Comment (harunurhan) [inline ui/src/literature/components/ArxivPdfDownloadAction.jsx]: `ExternalLink` here.

File: ui/src/literature/components/DOILinkAction.jsx [hunk 8]
```diff
@@ -1,30 +1,73 @@
...
+import DropdownMenu from '../../common/components/DropdownMenu';
+import IconText from '../../common/components/IconText';
+import ExternalLink from '../../common/components/ExternalLink';
 
 class DOILinkAction extends Component {
+  static renderLinkForDoi(doi) {
+    return `//doi.org/${doi.get('value')}`;
+  }
+
+  static renderDoi(doi) {
```
Comment (harunurhan) [inline ui/src/literature/components/DOILinkAction.jsx]: we need a `DOI` component that will be used also in `DOIList`. You can see the code duplication if you check that one.

This way actually you don't even need `EventTracker` specifically here.

File: ui/src/literature/components/DOILinkAction.jsx [hunk 9]
```diff
@@ -1,30 +1,73 @@
...
-          link={{
-            href,
-            target: '_blank',
-          }}
-        />
-      </EventTracker>
+      <ListItemAction>
+        <EventTracker eventId="DoiLink" eventPropName="onClick">
+          {dois.size > 1 ? (
+            <DropdownMenu
```
Comment (harunurhan) [inline ui/src/literature/components/DOILinkAction.jsx]: I would separate these two parts into `renderMultipleDOIDropdown`and `renderDOI`

File: ui/src/literature/components/LiteratureItem.jsx [hunk 10]
```diff
@@ -48,21 +49,27 @@ class LiteratureItem extends Component {
...
-                <ListItemAction
-                  iconType="login"
-                  text={`${citationCount} citations`}
-                  link={{ to: `${LITERATURE}?q=refersto:recid:${recordId}` }}
-                />
-              </EventTracker>
+              <ListItemAction>
+                <EventTracker eventId="Citations:Search">
+                  <Link
+                    className="no-transition ant-btn"
```
Comment (harunurhan) [inline ui/src/literature/components/LiteratureItem.jsx]: why is this necessary?

File: ui/src/literature/components/ArxivPdfDownloadAction.jsx (hunk 7 above)
Comment (monaawi) [reply ui/src/literature/components/ArxivPdfDownloadAction.jsx]: The ListItemAction was generating a button before, not an ExternalLink, I didn't want to change the behavior during the refactor

File: ui/src/literature/components/LiteratureItem.jsx (hunk 10 above)
Comment (monaawi) [reply ui/src/literature/components/LiteratureItem.jsx]: It was added in the ListItemAction before as well. It makes the citations link not be blue (unlike the other links in the result item) and not have a transition when you click on it.

File: ui/src/literature/components/DOILinkAction.jsx [hunk 11]
```diff
@@ -1,30 +1,66 @@
...
+        <IconText text={text} type={type} />
+      </DOILink>
+    );
+  }
+
   render() {
-    const { doi } = this.props;
-    const href = `//doi.org/${doi}`;
+    const { dois } = this.props;
+    const text = 'DOI';
```
Comment (harunurhan) [inline ui/src/literature/components/DOILinkAction.jsx]: no need to pass those, just have `static renderIconTextForDOI()` that returns  `<IconText text="DOI" type="link"/>` and use in both places.

File: ui/src/literature/components/DOILinkAction.jsx [hunk 12]
```diff
@@ -1,30 +1,66 @@
...
+      >
+        {this.renderDoiDropdownItem()}
+      </DropdownMenu>
+    );
+  }
+
+  renderDoiLink(text, type) {
+    const { dois } = this.props;
+    return (
+      <DOILink doi={dois.get(0).get('value')}>
```
Comment (harunurhan) [inline ui/src/literature/components/DOILinkAction.jsx]: `.getIn([0, 'value'])`

File: ui/src/literature/components/__tests__/DropdownMenu.test.jsx [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+import { Menu } from 'antd';
+import DropdownMenu from '../../../common/components/DropdownMenu';
+
+describe('DropdownMenu', () => {
+  it('renders', () => {
+    const wrapper = shallow(
+      <DropdownMenu title="title">
+        <Menu.Item key="1">
+          <li key="1">test</li>
+        </Menu.Item>{' '}
```
Comment (harunurhan) [inline ui/src/literature/components/__tests__/DropdownMenu.test.jsx]: remove `{' '}`

File: ui/src/common/containers/__tests__/__snapshots__/EditRecordActionContainer.test.jsx.snap [hunk 14]
```diff
@@ -3,69 +3,77 @@
 exports[`EditRecordActionContainer does not render edit button if user is not cataloger or superuser 1`] = `""`;
 
 exports[`EditRecordActionContainer renders edit button if user is cataloger 1`] = `
```
Comment (harunurhan) [inline ui/src/common/containers/__tests__/__snapshots__/EditRecordActionContainer.test.jsx.snap]: this isn't cool, should be a shallow render rather than `mount`. But I see that it's something PR has introduced, we can have a look together maybe after you're done with everything else.

File: ui/src/literature/components/DOILinkAction.jsx [hunk 15]
```diff
@@ -1,30 +1,62 @@
...
+import IconText from '../../common/components/IconText';
+import DOILink from './DOILink';
+import DOIMaterial from './DOIMaterial';
 
 class DOILinkAction extends Component {
+  static renderIconTextForDOI() {
+    return <IconText text="DOI" type="link" />;
+  }
+
+  renderDoiDropdownItem() {
```
Comment (harunurhan) [inline ui/src/literature/components/DOILinkAction.jsx]: Item**s**

File: ui/src/literature/components/DOILinkAction.jsx [hunk 16]
```diff
@@ -1,30 +1,62 @@
...
+      <Menu.Item key={doi.get('value')}>
+        <DOILink doi={doi.get('value')}>
+          {doi.get('value')}
+          <DOIMaterial material={doi.get('material')} />
+        </DOILink>
+      </Menu.Item>
+    ));
+  }
+
+  renderDoiDropdown() {
```
Comment (harunurhan) [inline ui/src/literature/components/DOILinkAction.jsx]: consistent naming would be better, so everwhere upper case `DOI`, except variable names.

File: ui/src/literature/components/__tests__/DOIMaterial.test.jsx [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+
+import DOIMaterial from '../DOIMaterial';
+
+describe('DOIMaterial', () => {
+  it('renders with material', () => {
+    const wrapper = shallow(<DOIMaterial material="myMaterial" />);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('renders without material', () => {
```
Comment (harunurhan) [inline ui/src/literature/components/__tests__/DOIMaterial.test.jsx]: `does not render`
